sqlmesh/core/model/definition.py

```diff
@@ -1876,6 +1880,28 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
+def _resolve_custom_session_properties(
+    defaults: t.Optional[t.Dict[str, t.Any]],
+    provided: t.Optional[exp.Expression] | t.Optional[t.Dict[str, t.Any]],
+) -> t.Optional[exp.Expression]:
+    if provided is not None and isinstance(provided, dict):
+        session_properties = {k: exp.Literal.string(k).eq(v) for k, v in provided.items()}
+    else:
+        session_properties = (
+            {expr.this.name: expr for expr in provided} if provided is not None else {}
+        )
+
+    if defaults is not None and defaults.get("session_properties") is not None:
```

Comment:
Nit:
```
if defaults and "session_properties" in defaults:
```

Reply:
yes, that was my preference too, I just thought you might have some convention I'm not aware of based on this comment 
![image](https://github.com/TobikoData/sqlmesh/assets/4933844/8826db9e-2987-4fef-aee3-8ee725288c09)